host/src/FunctionsNetHost/AppLoader/NetHost.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Runtime.InteropServices;
+
+namespace FunctionsNetHost
+{
+    internal class NetHost
+    {
+        [DllImport("nethost", CharSet = CharSet.Auto)]
+        private static extern int get_hostfxr_path(
+        [Out] char[] buffer,
+        [In] ref int buffer_size,
+        IntPtr reserved);
+
+        internal static string GetHostFxrPath()
+        {
+            char[] buffer = new char[500];
```

Comment:
Hopefully 500 is enough.
Example paths: 
C:\Program Files\dotnet\host\fxr\7.0.5\hostfxr.dll
C:\Program Files\dotnet\host\fxr\8.0.0-preview.4.23259.5\hostfxr.dll
/usr/lib/dotnet/host/fxr/7.0.5/libhostfxr.so

May be 200 may be enough.